### PR TITLE
fix(NUMA): 'default' is a keyword in some SDK languages

### PR DESCRIPTION
**Pull request by edwintorok**
'auto' is a keyword too.
Lets use 'default-policy' instead.

Fixes: d557b697a ("CP-38020: introduce Host.numa_affinity_policy")

### CODE REVIEW

**Comment by edwintorok** [comment]
I would've preferred 'default_policy' for the enum but lets be consistent with how 'best-effort ' and 'besteffort' looks like.

File: ocaml/idl/datamodel_host.ml [hunk 1]
```diff
@@ -1678,7 +1678,7 @@ let host_numa_affinity_policy =
            using soft-pinning, but the policy doesn't guarantee a balanced \
            placement, falling back to the 'any' policy."
         )
-      ; ( "default"
+      ; ( "default-policy"
```
Comment (robhoes) [inline ocaml/idl/datamodel_host.ml:1681]: Better use _ instead of -. That's the convention in the API. I see I've missed `best-effort`...

Comment (robhoes) [reply ocaml/idl/datamodel_host.ml:1681]: So let's also change `best-effort` to `best_effort`...

Comment (edwintorok) [reply ocaml/idl/datamodel_host.ml:1681]: We have 'balance-slb' that has a '-'. I'm fine either way

Comment (robhoes) [reply ocaml/idl/datamodel_host.ml:1681]: Then that one is bad also :) All enums in `host` and `VM` at least have _

Comment (robhoes) [reply ocaml/idl/datamodel_host.ml]: A bit too much of a replace-all now I think...

File: ocaml/xapi/xapi_xenops.ml [hunk 2]
```diff
@@ -593,11 +593,11 @@ module MD = struct
             RealTime (qos_class params)
         | "idle" ->
             Idle
-        | "best-effort" ->
+        | "best_effort" ->
```
Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml:596]: Too many things being replaced now...

Comment (edwintorok) [reply ocaml/xapi/xapi_xenops.ml:596]: ah we already have 'best-effort' elsewhere

Comment (robhoes) [reply ocaml/xapi/xapi_xenops.ml:596]: I'd just update the datamodel and let the compiler tell me what else to update.

**Review by robhoes — APPROVED** [review]

**Review by psafont — APPROVED** [review]

**Comment by psafont** [comment]
The commit [7de8c8f](https://github.com/xapi-project/xen-api/pull/5347/commits/7de8c8fdf82116225c4c2ec1babd10c4be56c02d) should be added as well

**Pull request merged**